# Présentation de la BD KITTI #

Contient 7481 images pour la phase d'entrainement et 7518 images pour la phase de test pour un total de +80K objets étiquetés.

Contenu de la BD KITTI : 
- Une BD qui contient les images couleur prises par une caméra RGB située à gauche de la voiture qui a fait l'acquisition ;
- Une BD qui contient les images couleur prises par une caméra RGB située à droite de la voiture ;
- Une BD qui contient les images en niveaux de gris prises par une caméra située à gauche de la voiture ;
- Une BD qui contient les images en niveaux de gris prises par une caméra située à droite de la voiture ;
- Une BD qui contient des nuages de points acquis par Velodyne ;
- Une BD qui contient les informations GPS.

On va étudier la BD qui contient les images couleur prises par la caméra gauche tout au long de ce script.

## Taille de la BD ##

In [14]:
import os
import glob
import imageio
import numpy as np

print(os.path.abspath("KITTI_database.ipynb"))
# Afin de connaître le chemin absolu pour pouvoir naviguer dans les dossiers ensuite

/Users/rblin/Documents/Untitled Folder/KITTI_database.ipynb


In [20]:
path = "/Users/rblin/Downloads"
os.listdir(path)

['.DS_Store',
 '.localized',
 'Chabot_Deep_MANTA_A_CVPR_2017_paper.pdf',
 'training',
 'PDE_2018_sans_trait_coupe.pdf',
 'data_object_image_2',
 'Anaconda3-5.2.0-MacOSX-x86_64.pkg',
 'CV.pdf']

### Exploration du dossier de training 

In [16]:
# Les fichiers sont des png du coup on va chercher le nombre d'images contenues dans le dossier

images_train = glob.glob(path+"/data_object_image_2/training/image_2/*.png")
print(len(images_train))

7481


### Exploration du dossier de test ###

In [13]:
images_test = glob.glob(path+"/data_object_image_2/testing/image_2/*.png")
print(len(images_test))

7518


On constate qu'on a 7481 images piur l'entraînement et 7518 images pour le test 

## Eléments composant la BD ##

On va maintenant étudier les éléments qui composent la BD (taille et nature de l'image)

In [17]:
# Etude de la première image du dossier à titre d'exemple

im = imageio.imread(images_train[0])
print(im.shape)

(375, 1242, 3)


On retrouve une image de taille 375*1242*3 ce qui correspond bien aux trois canaux de couleur de l'image.

In [25]:
label_files = os.listdir(path+"/training/label_2")

labels = list()
for label in label_files :
    fichier = open(path+"/training/label_2/"+label)
    contenu = fichier.read()
    lines = contenu.split("\n")
    for i in lines :
        elements = i.split(" ")
        if elements[0] not in labels :
            labels.append(elements[0])

print(labels)

# On retrouve ainsi toutes les étiquettes des éléments de la BD KITTI

Les classes sont définies dans cet ordre :
Cars, Vans, Trucks, Pedestrians, Sitters, Cyclists, Trams, Misc

['Car', 'Van', 'DontCare', '', 'Pedestrian', 'Truck', 'Cyclist', 'Misc', 'Tram', 'Person_sitting']


## Lien avec ImageNet ##

Maintenant que l'on a trouvé l'ensemble des catégories présentes dans le jeu de données KITTI, regardons si ces catégories sont présentes dans celles utiles pour réaliser les challenges sur ImageNet. Il faut savoir qu'ImageNet contient aujourd'hui plus de 14 000 000 d'images réparties en près de 22K catégories (site officiel d'ImageNet : http://www.image-net.org). Tous les ans une compétition est organisée sur une sélection de 1000 catégories parmi les 22K et les meilleurs résultats de l'état de l'art ont été obtenus à l'aide des réseaux de neurones NasNet et ResNet.

On retrouve ici la liste des 1000 catégories d'ImageNet qui servent au challenge : http://image-net.org/challenges/LSVRC/2012/browse-synsets

Les catégories de KITTI présentes dans cette liste sont :
- Car
- Van
- Truck

On constate que les piétons sont absents de cette liste ce qui représente un gros enjeux de la BD KITTI ce qui signifie que les réseaux entraînés sur cette BD ne sont pas les plus pertinents pour évaluer la BD KITTI. De plus, les éléments à repérer sont centrés dans l'image et sont souvent l'unique chose que contient l'image tandis que notre problème vise à reconnaître la présence ou non d'éléments dans une scène. 